src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -89,26 +84,41 @@ const loadAndRenderDefaultHeatmap = async () => {
   });
 };
 
-const stalledResponse = () => new Promise(() => { });
+const mockAPIResponses = generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID);
+
+const stalledResponse = () => delayedResponse({ body: 'Not found', status: 404 }, 4000);
```

Comment:
If it was me writing this, I don't think I would have `stalledResponse` based on `delayedResponse` function: it seems like too much effort for too little value of the abstraction. The code takes longer to understand -- one has to go back to `mockAPI` and look up `delayedResponse`, then has to come back here. And all that `delayedResponse` is, is resolving a promise after some time. I can easily imagine how people forget about this mock and don't use `delayedResponse` at all too. So I am not sure of the value of abstracting it.

Reply:
Right, I've replaced stalled response to just directly use `delayedResponse`